modules/services/git-workspace.nix

```diff
@@ -0,0 +1,153 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.git-workspace;
+  tomlFormat = pkgs.formats.toml { };
+  workspaceType = lib.types.submodule {
+    freeformType = tomlFormat.type;
+    options.provider = lib.mkOption {
+      type = lib.types.listOf (lib.types.submodule {
+        freeformType = tomlFormat.type;
+        options = {
+          provider = lib.mkOption {
+            type = lib.types.str;
+            example = "github";
+          };
+          name = lib.mkOption {
+            type = lib.types.str;
+            example = "torvalds";
+          };
+          path = lib.mkOption {
+            type = lib.types.path;
+            example = lib.literalExpression
+              ''"''${config.home.homeDirectory}/projects"'';
+          };
+        };
+      });
+    };
+  };
+in {
+  meta.maintainers = [ lib.maintainers.aciceri ];
+  options.services.git-workspace = {
+    enable = lib.mkEnableOption "git-workspace systemd timer";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = if config.programs.git-workspace.enable then
+        config.programs.git-workspace.package
+      else
+        pkgs.git-workspace;
+      defaultText = lib.literalExpression ''
+        if config.programs.git-workspace.enable then
+          config.programs.git-workspace.package
+        else
+          pkgs.git-workspace;
+      '';
```

Comment:
With removing the programs module this could be simplified, indeed the entire option could be defined using `mkPackageOption`.